### PR TITLE
feat: [PIE-2791]: ACR artifact implementation

### PR DESCRIPTION
**Pull request by Tamaraius**
##### Summary:

Implementation of ACR artifact

##### Jira Links:

https://harness.atlassian.net/browse/PIE-2791

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by Tamaraius** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts [hunk 1]
```diff
@@ -126,3 +143,13 @@ export interface ArtifactImagePathTagViewProps {
   tagDisabled: boolean
   isArtifactPath?: boolean
 }
+
+export interface ACRArtifactType {
+  identifier: string
+  tag: any
+  tagRegex: any
+  tagType: TagTypes
+  repository?: any
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:152]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9246757)
```suggestion
  repository?: unknown
```

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts [hunk 2]
```diff
@@ -126,3 +143,13 @@ export interface ArtifactImagePathTagViewProps {
...
+  subscription?: any
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:153]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9246757)
```suggestion
  subscription?: unknown
```

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts [hunk 3]
```diff
@@ -126,3 +143,13 @@ export interface ArtifactImagePathTagViewProps {
...
+  registry?: any
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:154]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9246757)
```suggestion
  registry?: unknown
```

**Comment by Tamaraius** [comment]
will create a new PR when Azure connector is merged

**Pull request closed without merging**